include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -4526,10 +4526,10 @@ inline DEAL_II_ALWAYS_INLINE
   // Cartesian cell
   if (!is_face && this->cell_type == internal::MatrixFreeFunctions::cartesian)
     {
-      for (unsigned int d = 0; d < dim; ++d)
-        for (unsigned int comp = 0; comp < n_components; ++comp)
+      for (unsigned int comp = 0; comp < n_components; ++comp)
+        for (unsigned int d = 0; d < dim; ++d)
           grad_out[comp][d] =
-            this->gradients_quad[(comp * dim + d) * nqp + q_point] *
+            this->gradients_quad[(comp * nqp + q_point) * dim + d] *
```

Comment:
Note that change here and in many places: I now place the `dim` components of the gradient to each component next to each other, which is still compatible with the fast evaluators. Unfortunately, I still have to keep the entries of different components grouped separately because the number of components is not known inside the evaluator, so adding that variable size would make the code considerably slower. I think this is a reasonable compromise. I did not do anything for the Hessian yet, so in case one could imagine similar steps there.

Reply:
I guess should we should add a change log entry (in the case someone accesses the buffers of FEval directly)?